src/sentry/api/event_search.py

```diff
@@ -855,21 +855,22 @@ def format_search_filter(term, params):
         project = None
         try:
             project = Project.objects.get(id__in=params.get("project_id", []), slug=value)
-        except Exception:
-            raise InvalidSearchQuery(
-                u"Invalid query. Project {} does not exist or is not an actively selected project.".format(
-                    value
+        except Exception as e:
+            if not isinstance(e, Project.DoesNotExist) or term.operator != "!=":
```

Comment:
I would move the term operator check up. If we are doing != on a project, there's no point doing the project lookup at all. We can save ourselves a DB call.

Reply:
I don't think will work when you use `All Projects`

Reply:
I confirmed that if you select `All Projects`, the lookup is still necessary to add the `!=` condition. I also added a test for that case.